src/walkway.js

```diff
@@ -324,8 +328,10 @@
   function getPolylineLength(polyline) {
     var dist = 0;
     var x1, x2, y1, y2;
+    var i;
```

Comment:
It does not, but variables get hoisted to the top of a function. So declaring all variables at the top is faster and a bit easier to understand in my opinion.
Also this prevents overwriting variables. For example, you got two `for` loops in one function with the same iteration name.